import "ydb/library/actors/protos/actors.proto";
import "ydb/core/protos/base.proto";
import "ydb/core/protos/bind_channel_storage_pool.proto";

package NKikimrTxScheme;
option java_package = "ru.yandex.kikimr.proto";

message TSchemeConfig {
    optional uint64 PipeClientCachePoolLimit = 1;
    //optional NKikimrTablet.TExecutorSettings ExecutorSettings = 2;
}

message TConfig {
    optional TSchemeConfig Current = 1;
    //optional NKikimrTxDataShard.TConfig DefaultDataShardConfig = 2; // DEPRECATED
    optional TSchemeConfig DefaultSchemeShardConfig = 3;
}

message TEvNavigateScheme {
    optional bytes Path = 1;
    optional bytes TextPath = 2;
    optional uint32 ReadMaterializedFamily = 3;
    optional bool ShowFamilies = 4 [default = true];
    optional bool ShowChildrenTags = 5 [default = true];
}

message TEvNavigateSchemePart {
    optional bytes Path = 1;
    optional bytes TextPath = 2;
    optional uint32 ReadMaterializedFamily = 3;
    optional NActorsProto.TActorId Source = 4;
    optional bool ShowFamilies = 5 [default = true];
    optional bool ShowChildrenTags = 6 [default = true];
}

message TEvNavigateSchemePartResult {
    optional uint32 Status = 1;
    optional uint64 ShardId = 2;
    optional bytes SchemePrefix = 3;
    optional bytes SchemePart = 4;
}

message TEvInitRootShard {
    optional NActorsProto.TActorId Source = 1;
    optional uint32 RootTag = 2;
    optional bytes RootTagName = 3;
    optional TConfig Config = 4;
    optional string Owner = 5;
    repeated NKikimrStoragePool.TStoragePool StoragePools = 6;
}

message TEvInitRootShardResult {
    optional uint64 Origin = 1;
    optional uint32 Status = 2; // defined in event
}

message TEvUpdateConfig {
    optional NActorsProto.TActorId Source = 1;
    optional TConfig Config = 2;
}

message TEvUpdateConfigResult {
    optional uint64 Origin = 1;
    optional NKikimrProto.EReplyStatus Status = 2;
    optional bytes Reason = 3;
}
